src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -0,0 +1,206 @@
+                var dataRemovingVisitor = new RemoveOptionalDataVisitor(optionallyEmittedData);
+                dataRemovingVisitor.Visit(sarifLog);
+            }
+
+            optionallyEmittedData = this.FilingContext.DataToInsert;
+            if (optionallyEmittedData != OptionallyEmittedData.None)
+            {
+                var dataInsertingVisitor = new InsertOptionalDataVisitor(optionallyEmittedData);
+                dataInsertingVisitor.Visit(sarifLog);
+            }
+
+            SplittingStrategy splittingStrategy = this.FilingContext.SplittingStrategy;
+            if (splittingStrategy == SplittingStrategy.None)
+            {
+                FileWorkItemsForSarifLog(sarifLog);
+                return;
+            }
+
+            for (int runIndex = 0; runIndex < sarifLog.Runs.Count; ++runIndex)
+            {
+                if (sarifLog.Runs[runIndex]?.Results?.Count > 0)
+                {
+                    IList<SarifLog> logsToProcess = new List<SarifLog>(new SarifLog[] { sarifLog });
+
+                    if (splittingStrategy != SplittingStrategy.PerRun)
+                    {
+                        SplittingVisitor visitor;
+                        switch (splittingStrategy)
+                        {
+                            case SplittingStrategy.PerRunPerRule:
+                                visitor = new PerRunPerRuleSplittingVisitor();
+                                break;
+                            
+                            case SplittingStrategy.PerRunPerTarget:
+                                visitor = new PerRunPerTargetSplittingVisitor();
+                                break;
+                            
+                            case SplittingStrategy.PerRunPerTargetPerRule:
+                                visitor = new PerRunPerTargetPerRuleSplittingVisitor();
+                                break;
+
+                            // TODO: Implement PerResult and PerRun splittings strategies
+                            //
+                            //       https://github.com/microsoft/sarif-sdk/issues/1763
+                            //       https://github.com/microsoft/sarif-sdk/issues/1762
+                            //
+                            case SplittingStrategy.PerResult:
+                            case SplittingStrategy.PerRun:                           
+                            default:
+                                throw new ArgumentOutOfRangeException($"GroupingStrategy: {splittingStrategy}");
+                        }
+
+                        visitor.VisitRun(sarifLog.Runs[runIndex]);
+                        logsToProcess = visitor.SplitSarifLogs;
+                    }
+
+                    for (int splitFileIndex = 0; splitFileIndex < logsToProcess.Count; splitFileIndex++)
+                    {
+                        SarifLog splitLog = logsToProcess[splitFileIndex];
+                        FileWorkItemsForSarifLog(sarifLog);
```

Comment:
>sarifLog [](start = 49, length = 8)

Should this not be the splitLog we just created rather than the original Sarif log?  If not, why do we create splitLog

Reply:
Yes, mile wide hole in the unit tests story, which I am closing today. Good catch. The ad hoc testing didn't catch this because these aren't performing any filtering or splitting currently.

---
In reply to: [380964011](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r380964011) [](ancestors = 380964011)